scipy/linalg/matfuncs.py

```diff
@@ -668,3 +675,329 @@ def rounded_sign(x):
+    False
+    """
+    A = _asarray_square(A)
+    return np.sum(A == A.T) == (A.shape[1]**2)
+
+
+def is_matrix_skew_symmetric(A, tol=1e-8):
+    """
+    Returns True if a matrix is skew symmetric
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is skew symmetric.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_skew_symmetric
+    >>> m = np.array([[5,1],[1,3]])
+    >>> print(is_matrix_skew_symmetric(m))
+    True
+    >>> m = np.array([[0, 2, -1],[-2, 0, -4],[-1, -4, 0]])
+    >>> print(is_matrix_skew_symmetric(m))
+    True
+    """
+    A = _asarray_square(A)
+    return (np.absolute(A - A.T) < tol).all()
+
+
+def is_matrix_nonsingular(A, tol=1e-8):
+    """
+    Returns True if a matrix is nonsingular
+    AKA invertible or nondegenerate
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is nonsingular.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_nonsingular
+    >>> m = np.array([[-1,1.5],[1,-1]])
+    >>> print(is_matrix_nonsingular(m))
+    True
+    >>> m = np.array([[1, -1, 0],[-1, 5, 0],[0, 0, 7]])
+    >>> print(is_matrix_nonsingular(m))
+    True
+    """
+    A = _asarray_square(A)
+    detA = det(A)
+    return (np.absolute(detA) >= tol).all()
+
+
+def is_matrix_singular(A, tol=1e-8):
+    """
+    Returns True if a matrix is singular
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is singular.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_singular
+    >>> m = np.array([[-1,1.5],[1.5,-1]])
+    >>> print(is_matrix_singular(m))
+    True
+    >>> m = np.array([[1, -1, 0],[-1, 5, 0],[0, 0, 7]])
+    >>> print(is_matrix_singular(m))
+    True
+    """
+    A = _asarray_square(A)
+    detA = det(A)
+    return (np.absolute(detA) < tol).all()
+
+
+def is_matrix_idempotent(A, tol=1e-8):
+    """
+    Returns True if a matrix is idempotent
+    Parameters
+    ----------
+    A : (N, N) array_like
+        Matrix to evaluate.
+    tol : float
+        Tolerance for the zero value.
+    
+    Returns
+    -------
+    X : boolean
+        True if a matrix A is idempotent.
+
+    Examples
+    --------
+    >>> from scipy.linalg import is_matrix_idempotent
+    >>> m = np.array([[3,-6],[1,-2]])
+    >>> print(is_matrix_idempotent(m))
+    True
+    >>> m = np.array([[2, -2, 4],[-1, 3, 4],[1, -2, -3]])
+    >>> print(is_matrix_idempotent(m))
+    False
+    """
+    A = _asarray_square(A)
+    return (np.absolute(np.matmul(A, A) - A) < tol).all()
```

Comment:
I don't think anyone needs this and I think too specific for scipy.

Reply:
removed